lib/dicon/secure_shell.ex

```diff
@@ -98,14 +104,18 @@ defmodule Dicon.SecureShell do
   def copy(conn, source, target) do
     %{size: size} = File.stat!(source)
     stream = File.stream!(source, [], div(size, 100))
-    {:ok, channel} = :ssh_sftp.start_channel(conn, [timeout: @timeout])
-    {:ok, handle} = :ssh_sftp.open(channel, target, [:write, :binary], @timeout)
-    Enum.each(stream, fn chunk ->
-      :ok = :ssh_sftp.write(channel, handle, chunk, @timeout)
-      IO.write "."
-    end)
-    IO.puts "\n"
-    :ok = :ssh_sftp.close(channel, handle, @timeout)
-    :ok = :ssh_sftp.stop_channel(channel)
+
+    with {:ok, channel} = :ssh_sftp.start_channel(conn, [timeout: @timeout]),
+         {:ok, handle} = :ssh_sftp.open(channel, target, [:write, :binary], @timeout),
```

Comment:
Not sure I understand what you mean :\ we're still matching on calls to `:ssh_sftp.close/3` and `:ssh_sftp.stop_channel/1` (the latter could be moved inside `do:` yes, but I think it's cleaner this way - we can change it however).